### PR TITLE
[FLINK-30657] Remove Shared and Key_Shared related tests in Pulsar connector

### PR DESCRIPTION
**Pull request by syhily**
## What is the purpose of the change

This pull request removes the flaky tests in flink-connector-pulsar. These tests have been dropped in the externalized flink-connector-pulsar and won't get fixed in the old Flink release. So we have discussed in https://issues.apache.org/jira/browse/FLINK-30413 and think these test codes are dead code now.

## Brief change log

Remove these test classes.

- SharedSplitAssignerTest.java
- PulsarUnorderedSourceReaderTest.java
- PulsarUnorderedPartitionSplitReaderTest.java
- PulsarSourceUnorderedE2ECase.java
- PulsarUnorderedSourceITCase.java

Drop Shared subscription test in PulsarSourceEnumeratorTest.java.

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (yes / no / don't know)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Review by tisonkun — APPROVED** [review]

File: flink-connector-pulsar/archunit-violations/f4d91193-72ba-4ce4-ad83-98f780dce581 [hunk 1]
```diff
@@ -10,9 +10,3 @@ org.apache.flink.connector.pulsar.source.PulsarSourceITCase does not satisfy: on
 * reside in a package 'org.apache.flink.runtime.*' and is annotated with @ExtendWith with class InternalMiniClusterExtension\
 * reside outside of package 'org.apache.flink.runtime.*' and is annotated with @ExtendWith with class MiniClusterExtension\
  or contain any fields that are public, static, and of type MiniClusterWithClientResource and final and annotated with @ClassRule or contain any fields that is of type MiniClusterWithClientResource and public and final and not static and annotated with @Rule
-org.apache.flink.connector.pulsar.source.PulsarUnorderedSourceITCase does not satisfy: only one of the following predicates match:\
```
Comment (XComp) [inline flink-connector-pulsar/archunit-violations/f4d91193-72ba-4ce4-ad83-98f780dce581:13]: That might be out of scope, but: There's another (empty) violations file. I haven't worked with archunit that much. I'm just wondering whether the empty file should be removed? :thinking:

**Review by XComp — APPROVED** [review]
I verified that the tests reported in FLINK-30413 are actually removed. :+1: I added two minor comments. Please find them below.

File: flink-connector-pulsar/archunit-violations/f4d91193-72ba-4ce4-ad83-98f780dce581 (hunk 1 above)
Comment (syhily) [reply flink-connector-pulsar/archunit-violations/f4d91193-72ba-4ce4-ad83-98f780dce581:13]: I don't know. I just leave these empty files. They are auto generated.

File: flink-connector-pulsar/src/test/java/org/apache/flink/connector/pulsar/source/enumerator/assigner/SharedSplitAssignerTest.java [hunk 2]
```diff
@@ -1,121 +0,0 @@
...
-import java.util.Optional;
-import java.util.Set;
-
-import static java.util.Collections.singletonList;
-import static org.assertj.core.api.Assertions.assertThat;
-import static org.junit.jupiter.api.Assertions.assertFalse;
-import static org.junit.jupiter.api.Assertions.assertTrue;
-
-/** Unit tests for {@link SharedSplitAssigner}. */
-class SharedSplitAssignerTest extends SplitAssignerTestBase {
```
[the remark this answers is not in the document]
Comment (syhily) [reply flink-connector-pulsar/src/test/java/org/apache/flink/connector/pulsar/source/enumerator/assigner/SharedSplitAssignerTest.java:41]: It's ok to refactor these tests. I have done in the main branch. So I prefer no to do too much refactor on old branches.

**Comment by tisonkun** [comment]
Thanks for your contribution @syhily! Mergiung...

**Pull request merged**